pulsar-transaction/coordinator/src/main/java/org/apache/pulsar/transaction/coordinator/impl/TxnLogBufferedWriterMetricsStats.java

```diff
@@ -0,0 +1,234 @@
+                pulsarBatchedLogEntrySizeBytesName,
+                k -> new Histogram.Builder()
+                        .name(pulsarBatchedLogEntrySizeBytesName)
+                        .labelNames(labelNames)
+                        .help("A metrics for how many bytes in per batch")
+                        .buckets(BYTES_SIZE_PER_ENTRY_BUCKETS)
+                        .register(registry));
+        pulsarBatchedLogEntrySizeBytesChild = pulsarBatchedLogEntrySizeBytes.labels(labelValues);
+
+        String pulsarBatchedLogOldestRecordDelayTimeSecondsName =
+                String.format("%s_batched_log_oldest_record_delay_time_seconds", metricsPrefix);
+        pulsarBatchedLogOldestRecordDelayTimeSeconds = (Histogram) COLLECTOR_CACHE.computeIfAbsent(
+                pulsarBatchedLogOldestRecordDelayTimeSecondsName,
+                k -> new Histogram.Builder()
+                        .name(pulsarBatchedLogOldestRecordDelayTimeSecondsName)
+                        .labelNames(labelNames)
+                        .help("A metrics for the max latency in per batch")
+                        .buckets(MAX_DELAY_TIME_BUCKETS)
+                        .register(registry));
+        pulsarBatchedLogOldestRecordDelayTimeSecondsChild =
+                pulsarBatchedLogOldestRecordDelayTimeSeconds.labels(labelValues);
+
+        String pulsarBatchedLogTriggeringCountByRecordsName =
+                String.format("%s_batched_log_triggering_count_by_records", metricsPrefix);
+        pulsarBatchedLogTriggeringCountByRecords = (Counter) COLLECTOR_CACHE.computeIfAbsent(
+                pulsarBatchedLogTriggeringCountByRecordsName,
+                k -> new Counter.Builder()
+                        .name(pulsarBatchedLogTriggeringCountByRecordsName)
+                        .labelNames(labelNames)
+                        .help("A metrics for how many batches were triggered due to threshold"
+                                + " \"batchedWriteMaxRecords\"")
+                        .register(registry));
+        pulsarBatchedLogTriggeringCountByRecordsChild = pulsarBatchedLogTriggeringCountByRecords.labels(labelValues);
+
+        String pulsarBatchedLogTriggeringCountBySizeName =
+                String.format("%s_batched_log_triggering_count_by_size", metricsPrefix);
+        pulsarBatchedLogTriggeringCountBySize = (Counter) COLLECTOR_CACHE.computeIfAbsent(
+                pulsarBatchedLogTriggeringCountBySizeName,
+                k -> new Counter.Builder()
+                        .name(pulsarBatchedLogTriggeringCountBySizeName)
+                        .labelNames(labelNames)
+                        .help("A metrics for how many batches were triggered due to threshold \"batchedWriteMaxSize\"")
+                        .register(registry));
+        pulsarBatchedLogTriggeringCountBySizeChild = pulsarBatchedLogTriggeringCountBySize.labels(labelValues);
+
+        String pulsarBatchedLogTriggeringCountByDelayTimeName =
```

Comment:
`bufferedwriter_flush_trigger_max_delay`

Reply:
I have renamed this metrics name suffix, thanks for suggestions :)